pkg/openwsn/include/openwsn_uart.h

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (C) 2018 Hamburg University of Applied Sciences
+ *               2020 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     pkg_openwsn
+ * @{
+ *
+ * For details on the implementation check pkg/openwsn/doc.txt
+ *
+ * @file
+ * @brief       RIOT adaption-specific definition of the "uart" bsp module.
+ *
+ * @author      Thomas Watteyne <watteyne@eecs.berkeley.edu>, February 2012
+ * @author      Peter Kietzmann <peter.kietzmann@haw-hamburg.de>, July 2018
+ *
+ * @}
+ */
+
+#ifndef OPENWSN_UART_H
+#define OPENWSN_UART_H
+
+#include "stdint.h"
+#include "board.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef STDIO_UART_DEV
+#define STDIO_UART_DEV              (UART_DEV(0))
+#endif
+/**
+ * @brief   OpenWSN default uart dev.
+ *
+ * @note    If only one UART is available that uart might collide with
+ *          STDIO_UART_DEV, otherwise the next available UART will be used.
+ *
+ */
+#ifndef OPENWSN_UART_DEV
+#ifdef MODULE_STDIO_NULL
+#define OPENWSN_UART_DEV            (STDIO_UART_DEV)
+#else
+#define OPENWSN_UART_DEV            ((STDIO_UART_DEV + 1) % UART_NUMOF)
+#endif
+#endif
+
+/**
+ * @brief   OpenWSN uart baudrate
+ */
+#ifndef OPENWSN_UART_BAUDRATE
+#ifndef STDIO_UART_BAUDRATE
+#define OPENWSN_UART_BAUDRATE       (115200U)
+#else
+#define OPENWSN_UART_BAUDRATE       (STDIO_UART_BAUDRATE)
+#endif
+#endif
+
+/**
+ * @brief   Initialize OpenWSN uart
+ *
+ * This will initialize a uart device at STDIO_UART_BAUDRATE/8N1. It will also
+ * initialize a timer to set sw tx_isr.
+ *
+ */
+void    uart_init_openwsn(void);
```

Comment:
are these symbol names dictated by openwsn? they seem awfully close to what RIOT's uart module might provide.

Reply:
This function `uart_init_openwsn` is the only one that is not, the other ones are dictated by `OpenWSN`. If your comment addressed mainly `uart_init_openwsn` then I can rename that one.

Reply:
no, that one has openwsn in it, probably won't collide.